corehq/apps/export/const.py

```diff
@@ -97,6 +97,9 @@
 # Used for automatically triggered exports
 SAVED_EXPORTS_QUEUE = 'saved_exports_queue'
 
+# If the queue contains this many tasks, log what's in it
+QUEUE_LENGTH_LOGGING_THRESHHOLD = 100
```

Comment:
My read of this code is that if the 100th export is queued, this will send a log message.  We may want this visibility if the queue continues to grow

You could log every time something is queued and the queue is above the threshold, but put a say 5 minute cache on that log function.  That way you wouldn't need to check the queue length every single time, and we'd continue getting information when the queue is above the threshold.

Not sure what the appropriate balance is there - conceivably you may want a different cache timeout for checking the queue length and for waiting between log entries.

Reply:
I think using a cache is a great idea.

The tricky part is that I'm less interested in the tasks that are being added to the queue, and more interested in the tasks that are holding up the queue. My goal here is to know whether adding workers/resources is all we need to do to resolve this issue, or whether adding resources won't fix this, and we should address one or more of the exports themselves. So I reckoned that a single snapshot would be enough (although we'd always get at least two, one on the way up, and the other on the way down).

Do you reckon what I'm doing here will allow me to achieve that goal? (And is it the right goal?)

Counting and dumping the tasks each takes at least 0.2s, so I've tried to do it as little as possible. But maybe adding caching and dropping the threshold can give us fresher data without compromising too much on speed.